include/IECore/MurmurHash.inl

```diff
@@ -334,6 +334,17 @@ inline void murmurHashAppend( IECore::MurmurHash &h, const std::vector<T> &data
 	h.append( data.data(), data.size() );
 }
 
+// vector<bool> is a special case because we can't get a pointer to the internal data,
+// because it's stored packed
+template<>
+inline void murmurHashAppend( IECore::MurmurHash &h, const std::vector<bool> &data )
+{
+	for( bool b : data )
+	{
+		h.append( b );
+	}
```

Comment:
We've already taken a different approach in `SharedDataHolder<std::vector<bool> >::hash()`, and we should make sure they match one way or another. Maybe the best compromise is to do the packing-into-bytes of `SharedDataHolder` but then hash those bytes one-by-one to avoid the memory allocation? Whatever we do we should remove the specialisation from `VectorTypedData.cpp`.

Reply:
If we want efficiency, then yes, we should pack the bytes.  Though if anyone really wanted efficiency they should avoid std::vector<bool> - if we packed bools ourselves into std::vector<char> or something then we could directly hash the packed representation.  My original assumption was that we just don't care about std::vector<bool>.  But if we do want to something not terrible, it does make sense to them ourselves, but into a unit larger than hashing bytes one-by-one.  I've written something that has a 128 bit buffer on the stack that it packs into - I haven't optimized it well, this sort of thing quite possibly calls for loop unrolling, but conceptually this approach should probably be possible to get close to optimal if we ever need to.  I probably should have more tests for this sort of bit-twiddling though.

Reply:
Oh, just realized something we both missed:  we need to hash in the length too when packing, otherwise hashing [1, 0] and [1, 0, 0, 0] would yield the same hash.

Reply:
Fixed squashed into 467a02b, and I did end up adding a test in a separate commit